document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java

```diff
@@ -86,38 +105,75 @@ private String getAccountLink(Accounts accounts, AccountType accountsType) {
         return accounts.getLinks().get(accountsType.getResourceKey());
     }
 
-    /**
-     * Creates the 'data' string in {@link DocumentInfoResponse}.
-     * @param transaction the transaction data
-     * @param abridgedAccountData the abridged accounts data
-     * @return data string in {@link DocumentInfoResponse}
-     */
-    private String createDocumentInfoResponseData(Transaction transaction, AbridgedAccountsApi abridgedAccountData) {
-        JSONObject abridgedAccountJSON = new JSONObject(abridgedAccountData);
-        JSONObject abridgedAccount = new JSONObject();
-        abridgedAccount.put("abridged_account", abridgedAccountJSON);
-        abridgedAccount.put("company_name", transaction.getCompanyName());
-        abridgedAccount.put("company_number", transaction.getCompanyNumber());
-        return abridgedAccount.toString();
-    }
-
     /**
      * Creates the {@link DocumentInfoResponse} object
      * @param transaction transaction data
      * @param accountType account type
-     * @param abridgedAccountData abridged account data
+     * @param accountData The account data
+     * @param mimeType
      * @return {@link DocumentInfoResponse} object
      */
-    private DocumentInfoResponse createResponse(Transaction transaction, AccountType accountType, AbridgedAccountsApi abridgedAccountData) {
+    private <T> DocumentInfoResponse createResponse(Transaction transaction, AccountType accountType,
+                                                    T accountData, String mimeType) throws ParseException {
         DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
-        documentInfoResponse.setData(createDocumentInfoResponseData(transaction, abridgedAccountData));
+        documentInfoResponse.setData(createDocumentInfoResponseData(transaction, accountData, accountType));
         documentInfoResponse.setAssetId(accountType.getAssetId());
         documentInfoResponse.setTemplateName(accountType.getTemplateName());
         documentInfoResponse.setPath(createPathString(accountType));
+
+        Map<String, String> descriptionValues = new HashMap<>();
+        descriptionValues.put("period_end_on", RESPONSE_DISPLAY_DATE_FORMAT.format(ISO_DATE_FORMAT.
+                parse(getCurrentPeriodEndOn(accountData))));
+
+        documentInfoResponse.setDescriptionValues(descriptionValues);
+        documentInfoResponse.setDescriptionIdentifier(accountType.getEnumerationKey());
+        documentInfoResponse.setContentType(mimeType);
```

Comment:
I thought the MIME type was going to be set to the MIME type header in the response from the data API request (e.g. abridged accounts API)? This could result in a different content type to the data type.